### ISSUE TITLE
Too much spacing for comment button on DateFields 

### ISSUE THREAD

**Issue by vorlif** [wagtail/wagtail#9543]
### Issue Summary

The comment button is directly behind the input field in all input fields. But for DateFields there is a large space between the input field and the comment button.

[Peek 2022-10-29 15-29.webm](https://user-images.githubusercontent.com/4333770/198834227-40aec86f-3b57-4e20-ba18-b011f8d965df.webm)

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. Create a page model with an `DateField`
    ```python
    class BlogPage(Page):
        date = models.DateField("Post date")
    ```
3. Migrate and start the server
4. Open den Wagtail admin interface
5. Enable the comment mode

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

-   Python version: 3.10.8
-   Django version: 4.1.1
-   Wagtail version: 4.0.4
-   Browser version: Firefox 106 and Chrome  107

**Comment by Theresa-o** [comment]
I'd reproduce this and

**Comment by Theresa-o** [comment]
@vorlif Thank you for taking the time to report this, I've been able to recreate the spacing between the `datefield` input and comment icon as seen below;

![wagcommentissuereproduced](https://user-images.githubusercontent.com/62052443/198839111-76c2764c-be9a-48cb-bef9-c8a5367d6b54.PNG)

Feedback would be provided shortly.

Thank you.

**Comment by vorlif** [comment]
@Theresa-o Gladly, thanks for your quick reply and your work!

**Comment by Theresa-o** [comment]
You're welcome

**Comment by lb-** [comment]
I am not sure this is a problem we should solve, it does seem odd when you have some 'short fields' but this feels like it  has been done intentionally.

<img width="787" alt="Screen Shot 2022-11-02 at 6 43 39 am" src="https://user-images.githubusercontent.com/1396140/199337269-5dec39f4-6571-48db-bed3-61d64c8c276a.png">

See how the checkbox also has this position (fixed to the right).

I think that maybe the chooser is the odd one out here in that it should also have the comment button to the right.

I will flag this as needs design decision before we move forward on a solution.

@Theresa-o  - sorry but we may need to put your PR on hold until we get a bit of clarity from others on the core team here.

**Comment by lb-** [comment]
Comment in UI team channel - https://wagtailcms.slack.com/archives/C0P6APKH9/p1667335617799779

**Comment by vorlif** [comment]
> I am not sure this is a problem we should solve, it does seem odd when you have some 'short fields' but this feels like it has been done intentionally.

Okay, I had only tested a few fields. :see_no_evil:  I just find it strange when I have to move my mouse over half the screen into the "nothing" to click a button there, which affects a field at the starting point. I know that there is a div element there, which probably triggers a hover effect. But I dare say that some users are confused why a button appears "back there". 

> I will flag this as needs design decision before we move forward on a solution.

I'm curious to see what the design says.

**Comment by lb-** [comment]
Thanks @vorlif  'needs design decision' is more about 'needs further discussion at the core team', not necessarily just UX design.

But yes, I agree it feels a bit odd but I also see the other perspective of a consistent 'position' for these buttons, almost like an invisible sidebar.

Much of this has changed though with the larger changes to the page editor so it may be worth revisiting this approach nonetheless.

**Comment by vorlif** [comment]
Thanks @lb- 

Yes, like an invisible sidebar you can see it too. Then, as you say, the choice field button would have to be moved, of course. But then you have an invisible sidebar with invisible buttons, which become visible when you act outside the sidebar. ^^

![image](https://user-images.githubusercontent.com/4333770/199346171-5eeb0ac6-b638-4ff8-9712-2e1f606d773c.png)

Anyway, I'll be happy with whatever you decide. You have also done a good job so far.

@Theresa-o Thanks for your work and sorry I didn't test more extensively before.

**Comment by Theresa-o** [comment]
@vorlif Thank you for reaching out.

**Comment by lb-** [comment]
PR that may have adjusted some of these positions https://github.com/wagtail/wagtail/pull/9072
I will also add this to the discussion about 3.*/4.* discussion as I think this more noticeable with the new editor UI. https://github.com/wagtail/wagtail/discussions/9553

**Comment by thibaudcolas** [comment]
Thanks for the report @vorlif!

I’ve had a chat about this with @benenright – the ideal implementation here is to have the comment buttons directly to the right of each widget wherever possible. This works well as-is for full-width widgets, and will require more work for the ones that aren’t full-width.

---

The changes needed are similar to what #9072 for a few types of widgets. Here is the full list of Django & Wagtail widgets we need to take into consideration:

```
AdminAutoHeightTextInput
Button
ButtonWithDropdown
ButtonWithDropdownFromHook
ButtonSelect
BooleanButtonSelect
AdminChooser (deprecated)
AdminPageChooser
AdminDateInput
AdminTimeInput
AdminDateTimeInput
FilteredSelect
SwitchInput
AdminTagWidget
DateRangePickerWidget
DraftailRichTextArea
StreamField
AdminImageChooser
AdminSnippetChooser
AdminDocumentChooser
AdminTaskChooser
TextInput
NumberInput
EmailInput
URLInput
PasswordInput
HiddenInput
DateInput
DateTimeInput
TimeInput
Textarea
CheckboxInput
Select
NullBooleanSelect
SelectMultiple
RadioSelect
CheckboxSelectMultiple
FileInput
ClearableFileInput
MultipleHiddenInput
SplitDateTimeWidget
SplitHiddenDateTimeWidget
SelectDateWidget
```

And here is a page model that has almost all of those set up: https://github.com/thibaudcolas/bakerydemo/blob/feature/everything-page/bakerydemo/ui_tests/models.py

An important point of consideration here is that although our default widgets all look very refined, technically under the hood we’re relying on Django’s own forms markup for almost all of them. So we have no control over the markup, and positioning things well can get tricky (for example we briefly tried to make widgets’ parent elements `flex` but even that felt potentially problematic).

And this is even more problematic for bespoke widgets, where we also have no control over the markup, and also no control of styles, and no way to test ahead of time within Wagtail itself.

For reference, screenshots of the comment positions as of Wagtail 4.1 (large files):

- https://user-images.githubusercontent.com/877585/199503878-a7c8b0ed-6096-489b-8d85-c0a8bae26c05.png
- https://user-images.githubusercontent.com/877585/199503890-615800ef-995b-475f-9e26-a90b073287fe.png

**Comment by lb-** [comment]
I will change this to an enhancement based on the comment above.

**Comment by thibaudcolas** [comment]
Fixed in #9549 for DateField, DateTimeField, TimeField. If there are other fields where we still don’t do this right please open a new ticket.
